### DISCUSSION TITLE
Menu variant

### DISCUSSION THREAD

**Discussion by robyn-bautista** [twilio-labs/paste#3144] [General] [answered]
Hi Paste team!

We are updating a menu dropdown to allow for quick updates to a status. We would like to apply a different color treatment to the menu dropdown to match badge colors, however it looks like we're unable to change the colors as this variant doesn't exist. We also attempted using the button instead, but it doesn't quite fit the styles we're looking for.

We'd like to propose adding a variant to the menu dropdown to accommodate - any advisement here? Thank you!

- Button vs menu 
- Button and button hack
- Original design

https://www.figma.com/file/4e366pjcsNYG0yohe4RzFU/P0-Final---Interaction-states-and-topics?node-id=1267%3A206479&t=smF2zY85tep1Su4D-1

**Comment by SiTaggart** [comment]
Hi @robyn-bautista 

Thanks for the question. We'd be curious to see the full context this component is designed to be used in, to give a better and more informed answer.

Some questions off the bat include;

- Where is this component used? In a form? Inside another component? Where on the page?
- What purpose does the color coding serve? What does the text not achieve on its own?
- Have you explored using a badge?

Thanks

**Reply by robyn-bautista** [reply]
It is used in the header for quick access. The color is decorative to emphasize state, to your point in place of the badge. There needs to be an easy way to change status via the header without diving into the "more" menu

**Reply by robyn-bautista** [reply]
in the screenshot above there would eventually be four statuses to choose from

**Reply by serifluous** [reply]
Hey @robyn-bautista ! As part of the Navigation UI Kit that's on our roadmap, we'll be composing a badge menu like this ([see figma here](https://www.figma.com/file/KUOgqFmTe46GOAvQNOhRnY/Navigation-UI-Kit?node-id=993-269617&t=AIWNbq38cs56lQ6p-4)):

<img width="268" alt="Screenshot 2023-04-14 at 12 00 27 PM" src="https://user-images.githubusercontent.com/3821840/232133456-e06d13ce-5781-474e-bfb9-db7436ef0670.png">

The badge will stay that gray color as the statuses change so we can keep the badge itself scalable for more and different statuses that products might want to use in it. The main thing that changes is the icon + icon color + text content in the badge. I'd recommend using the same for your use case in case you ever need to add more types of statuses than you have right now. Check out the Status pattern for process icons you could use: https://paste.twilio.design/patterns/status#process-status. Does something like this work for your use case?

You could also consider using the different badge variants we have here, but you'll be more limited by the variants we have compared to using icons to indicate different statuses: https://paste.twilio.design/components/badge

**Reply by robyn-bautista** [reply]
Got it, this is helpful, thank you! So if I'm understanding correctly, 1 + 2 would be in line with current components (and the status menu being built):

https://www.figma.com/file/4e366pjcsNYG0yohe4RzFU/P0-Final---Interaction-states-and-topics?node-id=1267%3A206479&t=smF2zY85tep1Su4D-1

**Reply by serifluous** [reply]
Yep, 1 and 2 are doable with the current components. 1 would be the recommended way to do it since 2 is a lil too box-within-a-box-y.

We're also going to be creating a variant of the badge with a white instead of gray background if that works better for you: https://www.figma.com/file/KUOgqFmTe46GOAvQNOhRnY/Navigation-UI-Kit?node-id=993-267622&t=eTgfNgdDiME5gsuk-4

**Reply by robyn-bautista** [reply]
Thank you! when is the badge menu slated to be built/released? 

I'm also wondering if we could still potentially bring what was done in option 1 to option 2 - we could potentially do some testing on our end with the original option versus icons in option 1

**Reply by loreina** [reply, marked as answer]
Hi @robyn-bautista, you can create this status badge using our components today.

Here's a barebones codesandbox example of what it would look like: https://codesandbox.io/s/bold-lalande-ddo8nn?file=/src/index.tsx. We'll add a status badge example to our docs site later as well.

Hope this helps!
